From 05966a7018a4a395179a8d4fb1c66aa14d57c22a Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:07:49 -0400
Subject: [usb] storage/shuttle_usbat: fix discs being detected twice

Message-id: <1410887463-20674-398-git-send-email-dzickus@redhat.com>
Patchwork-id: 95430
O-Subject: [RHEL7 PATCH 397/591] usb: storage: shuttle_usbat: fix discs being detected twice
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit df602c2d2358f02c6e49cffc5b49b9daa16db033
Author: Daniele Forsi <dforsi@gmail.com>
Date:   Tue Apr 29 11:44:03 2014 +0200

    usb: storage: shuttle_usbat: fix discs being detected twice

    Even if the USB-to-ATAPI converter supported multiple LUNs, this
    driver would always detect the same physical device or media because
    it doesn't use srb->device->lun in any way.
    Tested with an Hewlett-Packard CD-Writer Plus 8200e.

    Signed-off-by: Daniele Forsi <dforsi@gmail.com>
    Cc: stable <stable@vger.kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/shuttle_usbat.c b/drivers/usb/storage/shuttle_usbat.c
index 4ef2a80..008d805 100644
--- a/drivers/usb/storage/shuttle_usbat.c
+++ b/drivers/usb/storage/shuttle_usbat.c
@@ -1851,7 +1851,7 @@ static int usbat_probe(struct usb_interface *intf,
  us->transport_name = "Shuttle USBAT";
  us->transport = usbat_flash_transport;
  us->transport_reset = usb_stor_CB_reset;
- us->max_lun = 1;
+ us->max_lun = 0;
 
  result = usb_stor_probe2(us);
  return result;
-- 
1.7.1